server/routerlicious/packages/routerlicious-base/src/alfred/routes/api/api.ts

```diff
@@ -144,14 +145,24 @@ export function create(
 		async (request, response) => {
 			const tenantId = getParam(request.params, "tenantId");
 			const documentId = getParam(request.params, "id");
-			const signalContent = getParam(request.body, "singalContent");
 			try {
-				const signalRoom: IRoom = { tenantId, documentId };
-				const payload: IBroadcastSignalEventPayload = { signalRoom, signalContent };
-				collaborationSessionEventEmitter.emit("broadcastSignal", payload);
-				response.status(200).send("OK");
+				const signalContent = JSON.parse(
+					getParam(request.body, "signalContent"),
+				) as IRuntimeSignalEnvelope;
```

Comment:
The `as` doesn't do any actual validation, let's add explicit checks for `typeof signalContent?.contents?.type === "string"`, `signalContents.contents.content !== undefined`. I might have gotten the paths wrong :).

Also, I think it's cleaner to limit the new try-catch block to the `JSON.parse()` and the new checks, and `return` immediately after setting the `response` properties in the catch block to avoid moving past that point (I can't recall if there's a particular pattern used in Express routes for that or a simple `return;` is enough).